Instant speed changes

TheCycoONE opened the pull request:
Avoid waiting for next game hour to adjust to speed changes.

Fixes #2758

Alberth289346 commented:
Approach seems fine to me.

Optionally:  
Method could use cleanup of the names, it has `current` values, `old` values, and just `values`.
In reality however there are just 2 sets of values. It would help reading the code if each set of variables has a distinct common prefix.

ARGAMX commented:
Wow! Speed ​​switches instantly now! Cool! Great job! 🔥

But I noticed that when loading some savegames, but not all, the following error occurs:

<img width="346" alt="Screenshot 2025-01-19 at 13 43 34" src="https://github.com/user-attachments/assets/454a5d83-1986-41da-8f99-0cac829f0326" />

Here is an example of a save that I can't load after applying the change to the current master:

[speed_switch_test.sav.zip](https://github.com/user-attachments/files/18468536/speed_switch_test.sav.zip)

ARGAMX commented:
Also, maybe this is not directly related to this PR and task, but I would like to draw attention to the fact that if the X key was held down during saving (especially autosaving) to speed up the game speed, then when loading such a save, this speed will be keeped. Which looks a little strange after loading and you want to immediately slow down the speed of that chaos. This became especially relevant after #2688, in which the X key speed seemed to about double compared to what it was before In 0.68.0.

Here is such save:

[Autosave5_ultraspeed.sav.zip](https://github.com/user-attachments/files/18468630/Autosave5_ultraspeed.sav.zip)

TheCycoONE commented:
hmm... are those all older saves? I would like one if you can provide it here.

TheCycoONE commented:
Try again, I addressed the naming and the speed key on load issue. It's not clear how this could cause errors on load but I added some paranoia handling of missing globals as well.

lewri reviewed (commented):
I'm not sure of the increased use of CamelCase but logically it all seems fine to me too.

ARGAMX commented:
> Try again, I addressed the naming and the speed key on load issue. It's not clear how this could cause errors on load but I added some paranoia handling of missing globals as well.

I just test again and seems both points resolved.
All savegames loading successfully now.
Speed key now reseting after load.
Thanks! 👍

Works good to me.

TheCycoONE commented:
Well the lua variable conventions don't match the C++ ones at all do they... fixed.
